shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/DriverDataSourceCache.java

```diff
@@ -44,23 +43,17 @@ public DataSource get(final String url) throws SQLException {
         if (dataSourceMap.containsKey(url)) {
             return dataSourceMap.get(url);
         }
-        DataSource dataSource;
+        return dataSourceMap.computeIfAbsent(url, DriverDataSourceCache::createDataSource);
```

Comment:
This may impact concurrent performance in Java 8. https://bugs.openjdk.org/browse/JDK-8161372

Reply:
Yep, this problem is that when the key present，Therefore, keeping the above code can avoid this problem。

```
if (dataSourceMap.containsKey(url)) {
    return dataSourceMap.get(url);
}
```
Similar solutions
https://github.com/mybatis/mybatis-3/pull/1934/commits/beee143f72b0c396591aa0c59739a1e64fd41867

Reply:
Generally, we don't need this code ``` if (dataSourceMap.containsKey(url)) ```, but because of this problem [JDK-8161372](https://bugs.openjdk.org/browse/JDK-8161372), we keep it.

Reply:
Got it. I didn't saw the logic above.